packages/wrangler/src/versions/deploy.ts

```diff
@@ -205,6 +209,61 @@ function getConfig(
 	return config;
 }
 
+/**
+ * Prompts the user for confirmation when overwriting the latest deployment, given that it's split.
+ */
+export async function confirmLatestDeploymentOverwrite(
+	accountId: string,
+	scriptName: string
+) {
+	try {
+		const latest = await fetchLatestDeployment(accountId, scriptName);
+		if (latest && latest.versions.length >= 2) {
+			// Fetch the version details.
+
+			const versionCache: VersionCache = new Map();
+			const versionIds = latest.versions.map((v) => v.version_id);
+			await fetchVersions(accountId, scriptName, versionCache, ...versionIds);
+
+			const versions = latest.versions.map((v) => {
+				const version = versionCache.get(v.version_id);
+				assert(version);
+				return version;
+			});
+
+			const traffic = new Map(
+				latest.versions.map((v) => [v.version_id, v.percentage])
+			);
+
+			// Print message and confirmation.
+
+			cli.warn(
+				`Your last deployment has multiple versions. To progress this deployment use "wrangler versions deploy" instead.`,
+				{ shape: cli.shapes.corners.tl, newlineBefore: false }
+			);
+			cli.newline();
+			cli.logRaw(
+				`${leftT} Your last deployment has ${versions.length} version(s):`
+			);
+			printVersions(versions, traffic);
+
+			return inputPrompt<boolean>({
+				type: "confirm",
+				question: `"wrangler deploy" will upload a new version and deploy it globally immediately.\nAre you sure you want to continue?`,
+				label: "",
+				defaultValue: true, // defaults to true to keep current behavior.
+				acceptDefault: !isInteractive() || CI.isCI(),
```

Comment:
Thanks for calling out this default is for back-compat

Really this default should be `false` as we're trying to prevent a footgun. But `false` default means we're half-preventing the footgun in interactive mode but not in CI (for back-compat)

I would suggest the following, which would make the default false in interactive mode to prevent a user in autopilot hitting enter and instead needing to explicitly respond yes. And still default to true in CI for back-compat.
```suggestion
				defaultValue: !isInteractive() || CI.isCI(), // defaults to true in CI for back-compat
				acceptDefault: !isInteractive() || CI.isCI(),
```